dbms/src/Client/Benchmark.cpp

```diff
@@ -320,17 +340,60 @@ class Benchmark
 			<< "result MiB/s: " << (info.result_bytes / seconds / 1048576) << "."
 			<< "\n";
 
-		for (size_t percent = 0; percent <= 90; percent += 10)
+		auto print_percentile = [&](double percent)
+		{
 			std::cerr << percent << "%\t" << info.sampler.quantileInterpolated(percent / 100.0) << " sec." << std::endl;
+		};
 
-		std::cerr << "95%\t" 	<< info.sampler.quantileInterpolated(0.95) 	<< " sec.\n";
-		std::cerr << "99%\t" 	<< info.sampler.quantileInterpolated(0.99) 	<< " sec.\n";
-		std::cerr << "99.9%\t" 	<< info.sampler.quantileInterpolated(0.999) 	<< " sec.\n";
-		std::cerr << "99.99%\t" << info.sampler.quantileInterpolated(0.9999) << " sec.\n";
-		std::cerr << "100%\t" 	<< info.sampler.quantileInterpolated(1) 		<< " sec.\n";
+		for (int percent = 0; percent <= 90; percent += 10)
+			print_percentile(percent);
+		print_percentile(95);
+		print_percentile(99);
+		print_percentile(99.9);
+		print_percentile(99.99);
 
 		info.clear();
 	}
+
+	void reportJSON(Stats & info, const std::string & filename)
+	{
+		std::ofstream jout(filename);
```

Comment:
Let use WriteBufferFromFile and DB/IO/Operators.h
By the way, it truncates existing file.